File: app/src/main/java/com/github/anonisnap/fragmentviewdemo/ui/sub_page_one/SubPageOneViewModel.java
package com.github.anonisnap.fragmentviewdemo.ui.sub_page_one;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class SubPageOneViewModel extends AndroidViewModel {

	private final MutableLiveData<String> mText;

	public SubPageOneViewModel(Application application) {
		super(application);
		mText = new MutableLiveData<>();
		mText.setValue("This is the First Sub Page");
	}

	public LiveData<String> getText() {
		return mText;
	}
}

File: app/src/main/java/com/github/anonisnap/fragmentviewdemo/ui/sub_page_one/SubPageOneViewModel.kt
package com.github.anonisnap.fragmentviewdemo.ui.sub_page_one

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData

class SubPageOneViewModel(application: Application?) : AndroidViewModel(application!!) {
    private val mText: MutableLiveData<String?>
    val text: LiveData<String?>
        get() = mText

    init {
        mText = MutableLiveData()
        mText.value = "This is the First Sub Page"
    }
}